### PR TITLE
feat: custom stop on v1 (CORE-5457)

### PR DESCRIPTION
**Pull request by leartgjoni-voiceflow**
to-do:
- update general-types

[general-types](https://github.com/voiceflow/general-types/pull/59)
[runtime-client-js](https://github.com/voiceflow/runtime-client-js/pull/52)

### CODE REVIEW

File: lib/services/runtime/handlers/_v1.ts [hunk 1]
```diff
@@ -42,12 +43,15 @@ export const _V1Handler: HandlerFactory<Node, typeof utilsObj> = (utils) => ({
 
     runtime.trace.addTrace<TraceFrame>({
       type: node.type,
-      payload: { data: node.payload, paths: node.paths },
+      payload: { data: node.payload, paths: node.paths, stop: node.stop, defaultPath: node.defaultPath },
     });
 
+    const stopTypes = runtime.storage.get<string[]>(StorageType.STOP_TYPES) || [];
+
+    const stop = stopTypes.includes(_.get(node.payload, 'name')) || node.stop;
```
Comment (DecathectZero) [inline lib/services/runtime/handlers/_v1.ts:51]: I'm curious why we have to use _.get instead of `node.payload?.name`?

**Review by DecathectZero — APPROVED** [review]

File: lib/services/runtime/types.ts [hunk 2]
```diff
@@ -18,11 +19,12 @@ export const isIntentRequest = (request: RuntimeRequest): request is IntentReque
...
 
 export const isRuntimeRequest = (request: any): request is RuntimeRequest => {
-  return request === null || !!(typeof request.type === 'string' && !!request.type && request.payload);
+  return request === null || !!(typeof request.type === 'string' && !!request.type);
 };
 
-export const isGeneralRequest = (request: RuntimeRequest): request is Request<string, { name: string }> => {
-  return !!(request as Request<string, { name: string }>)?.payload.name;
+type GeneralRequest = Request<string, undefined | {}>;
+export const isGeneralRequest = (request: RuntimeRequest): request is GeneralRequest => {
```
Comment (DecathectZero) [inline lib/services/runtime/types.ts:26]: I don't even know if it's the best to do a `isGeneralRequest` check anymore - I think it's okay for the request to contain data. We simply just assume every request is a general request if it doesn't match anything else prior (like intent request?)

File: lib/services/runtime/handlers/_v1.ts (hunk 1 above)
Comment (leartgjoni-voiceflow) [reply lib/services/runtime/handlers/_v1.ts:51]: `payload` is of type `unknown`

File: lib/services/runtime/handlers/event/index.ts [hunk 3]
```diff
@@ -28,9 +28,9 @@ export const generalEventMatcher = {
     event: Event | null;
   }): context is { runtime: Runtime<Request>; event: Event<string, GeneralEvent> } => {
     const request = context.runtime.getRequest();
-    if (!isGeneralRequest(request)) return false;
+    if (!request || isIntentRequest(request)) return false;
```
Comment (leartgjoni-voiceflow) [inline lib/services/runtime/handlers/event/index.ts:31]: if intent request - don't match. Example, if stopped on an interaction block and an intent comes in, without this check the `generalEventMatcher` will return `true` thus matching the first path in the interaction node no matter what.

**Pull request merged**